src/query/api/v1/httpd/handler.go

```diff
@@ -131,8 +132,12 @@ func (h *Handler) RegisterRoutes() error {
 	).Methods(openapi.HTTPMethod)
 	h.router.PathPrefix(openapi.StaticURLPrefix).Handler(logged(openapi.StaticHandler()))
 
+	timeoutOpts := &prometheus.TimeoutOpts{
+		FetchTimeout: h.embeddedDbCfg.Client.FetchTimeout,
```

Comment:
@r and @richardartoul are we guaranteed that `embeddedDbCfg.Client != nil` here?

Reply:
Rather than having this, have the handler take in a `TimeoutOpts` in `NewHandler` and set it on the Handler stuct

Reply:
I don't think its guaranteed to be set

Reply:
Okay maybe I'll just set it here then if it's nil